quarkchain/evm/messages.py

```diff
@@ -137,6 +135,23 @@ def validate_transaction(state, tx):
     if not tx.sender:  # sender is set and validated on Transaction initialization
         raise UnsignedTransaction(tx)
 
+    if tx.version == 2:
+        if (
```

Comment:
I am thinking in the case of tx.version == 2, we should set tx.network = eth_chain_id so that it will be much easier to diagnoses the content of a unserialized tx (and thus the tx is more like a superset of EIP155 tx).